plugin/sampling/strategystore/static/strategy_store_test.go

```diff
@@ -381,6 +382,7 @@ func TestAutoUpdateStrategyWithFile(t *testing.T) {
 
 func TestAutoUpdateStrategyWithURL(t *testing.T) {
 	mockServer, mockStrategy := mockStrategyServer()
+	defer mockServer.Close()
```

Comment:
a better fix would be to pass `t` above: `mockStrategyServer(t)` and inside that function add
```
t.Cleanup(func() { mockServer.Close() })
```
This way you don't have to keep cleaning it up everywhere

Reply:
I've implemented cleanup approach into tests and removed redundant cleanups